[Shipping labels] Add "id" parameter to purchase label endpoint

irfano opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses, e.g., WOOMOB-373. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR adds the required "id" parameter, which represents the shipment index, to the `POST /wcshipping/v1/label/purchase/{order_id}` endpoint. Related discussion: p1749620015063859-slack-C05VBLKHHV1

I added an additional fix to this PR (723d73a7658193670a160027e37ae59928e6bce9). The refund screen relies on config data from the cache, but we weren't updating the cached config after a successful purchase. This caused an incorrect UI state on the refund screen. In this commit, we update the cached config using the purchased label response.

### Steps to reproduce
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->
1. Open an order in the app.
2. Tap the **Create shipping label** button.
3. Fill in address fields if missing.
4. Select a package and rate.
5. Tap the **Purchase label** button.
6. Restart the app, open the same order, and compare purchased shipment with the **Create shipping labels** page on the web.

### The tests that have been performed
<!-- To give the reviewer an idea of what could be missed in terms of testing -->
Steps above

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

irfano committed:
Pass shipment ID when purchasing labels

irfano committed:
Remove unnecessary parameter, async from purchase label endpoint

irfano committed:
Fix WooShippingLabelCreationViewModelTest

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/261">22.7</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

irfano committed:
Update config cache when label status changes

Updates the `WooShippingConfigDataStore` with the latest shipping label information when the status of a label being purchased transitions from `PURCHASE_IN_PROGRESS` to `PURCHASED`. This ensures that the UI reflects the most up-to-date label data after a successful purchase.

hichamboushaba reviewed (approved):
Looks good, thanks @irfano
